zerver/views/push_notifications.py

```diff
@@ -58,52 +59,53 @@ def validate_token(token_str: str, kind: int) -> None:
 
 
 @human_users_only
-@has_request_variables
+@typed_endpoint
 def add_apns_device_token(
     request: HttpRequest,
     user_profile: UserProfile,
-    token: str = REQ(),
-    appid: str = REQ(str_validator=check_app_id),
+    *,
+    token: str,
+    appid: Annotated[str, AfterValidator(lambda x: check_app_id("appid", x))],
```

Comment:
No need to use `AfterValidator`, pydantic provides a way to check for regex using `StringConstraints`.

`StringConstraints(pattern="^[.a-zA-Z0-9-]+$")`.